jep/0000/README.adoc

```diff
@@ -0,0 +1,345 @@
+//| Superseded-By
+//| :bulb: JEP-NUMBER :bulb:
+//
+//
+// Uncomment when this JEP status is set to Accepted, Rejected or Withdrawn.
+//| Resolution
+//| :bulb: Link to relevant post in the jenkinsci-dev@ mailing list archives :bulb:
+
+|===
+
+== Abstract
+
+This JEP introduces a new system, which enables Jenkins plugin maintainers to add Continuous 
+Delivery to their projects. It is:
+
+* Opt-in: plugin maintainers can choose to use this
+* Simple to configure: very little needs to be changed in order to enable CD
+* Trusted CI/CD: deployments are produced from a single source of truth, which is managed by 
+the Jenkins project
+
+== Specification
+
+=== Overview
+
+We'll use a short story to illustrate what Autorelease will look like when a plugin maintainer 
+named Karl decides that he wants to perform continuous delivery of his (fictional) Karl-Plugin.
+
+To do so, Karl will:
+
+1. Verify support. Karl's plugin is already in the 
+link:https://github.com/jenkinsci[jenkinsci Organization on GitHub], and is 
+using `buildPlugin` when built on ci.jenkins.io.
+
+2. Choose a branch from which to release. Karl decides he wants to release from the `master` branch, 
+as that is the simplest.
+
+3. Settle on a version numbering scheme. Becuase it's in fairly wide use across the Jenkins 
+ecosystem, Karl settles on link:https://semver.org/[semantic versioning].
+
+4. Add a file to the top directory of his repository called `Autorelease.yaml`. 
+
+5. A webhook is configured on Karl's repository on GitHub to point to `{the-webhook-receiver}`
+
+Once ready to create a new release of his plugin, Karl merges some changes to the `master` branch. 
+At which point, the following takes place under the covers: 
+
+1. A webhook is sent to `{the-webhook-receiver}` by GitHub
+
+2. The receiver forwards this hook to `${trusted-ci-server}`, which is protected by a VPN
+
+3. `${trusted-ci-server}` uses the enhanced version of `buildPlugin` to validate the correctness of 
+Karl's `Autorelease.yaml` file, and the consistency of Karl's chosen version numbering.
+
+4. Once `Autorelease.yaml` has been validated, a plugin build is performed on `${trusted-ci-server}` 
+as one would expect.
+
+5. After the build is successful, `${trusted-ci-server}` handles deployment of the built plugin to 
+Artifactory and the Jenkins Update Center.
+
+=== Detailed Setup
+
+==== Autorelease.yaml
+
+A YAML markerfile, stored in the plugin's own repository. Absence of this markerfile will 
+be functionally equivalent to the plugin maintainer opting out of Autorelease. 
+
+In the following example, we'll stick with plugin developer Karl, and his fictional 
+plugin `karl-plugin`. Karl's `Autorelease.yaml` file looks like this:
+
+```
+plugin: karl
+kind: plugin
+metadata:
+  autoreleaseEnabled: true
+  triggeringBranch: master
+  versionNumberFields: 3
+```
+
+In the above example `Autorelease.yaml`, we can see the following:
```

Comment:
[Callouts](https://www.methods.co.nz/asciidoc/chunked/ch20.html) would help here.

Reply:
That's great. Didn't know that was a thing. :1st_place_medal:

Reply:
This might need some tweaking. It's doing "something" but it's not as pretty looking as I was expecting it to be:

![image](https://user-images.githubusercontent.com/21689198/59532884-f14c9c80-8eb7-11e9-841c-477e4e357cd6.png)

Reply:
I have no idea offhand if GitHub’s renderer supports them.

Reply:
It's still a pretty neat thing, though, thanks for showing that to me.